extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java

```diff
@@ -504,8 +504,11 @@ public static Map<OidcEndpoint.Type, List<OidcRequestFilter>> getOidcRequestFilt
             for (OidcRequestFilter filter : container.listAll(OidcRequestFilter.class).stream().map(handle -> handle.get())
                     .collect(Collectors.toList())) {
                 OidcEndpoint endpoint = ClientProxy.unwrap(filter).getClass().getAnnotation(OidcEndpoint.class);
-                OidcEndpoint.Type type = endpoint != null ? endpoint.value() : OidcEndpoint.Type.ALL;
-                map.computeIfAbsent(type, k -> new ArrayList<OidcRequestFilter>()).add(filter);
+                OidcEndpoint.Type[] types = endpoint != null ? endpoint.value()
+                        : new OidcEndpoint.Type[] { OidcEndpoint.Type.ALL };
```

Comment:
Perhaps check if `endpoint != null` to avoid creating this array every time this method is called?

Reply:
@gastaldi Sure, do you mean to have a static final property `OidcEndpoint.Type[] ALL_TYPES = new OidcEndpoint.Type[] { OidcEndpoint.Type.ALL }` and use that when endpoint is null ? Sorry if I'm missing your suggestion

Reply:
@gastaldi Did just that, not sure if we can optimize further, `map.computeIfAbsent(type, k -> new ArrayList<OidcRequestFilter>()).add(filter);` is done only the first time I believe when a filter is added to a List corresponding to a given type, ex, there could be many filters which apply to `ALL` endpoints